File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.py
point = list(map(int, input().split()))

if (point[0] < 0 or point[0] >= point[2] or point[1] < 0 or point[1] >= point[3] or
        point[2] < 0 or point[2] > 1000 or point[3] < 0 or point[3] > 1000):
    print("error")
else:
    print(min(point[0], point[2] - point[0], point[1], point[3] - point[1]))

File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.java
import java.util.Scanner;

import static java.lang.Math.min;

public class Main {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int x = scan.nextInt();
        int y = scan.nextInt();
        int w = scan.nextInt();
        int h = scan.nextInt();

        if (x < 0 || x >= w || y < 0 || y >= h || w < 0 || w > 1000 || h < 0 || h > 1000)
            System.out.println("error");
        else
            System.out.println(min(x, min(w - x, min(y, h - y))));
    }
}